lib/facility.rb

```diff
@@ -1,14 +1,68 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def determine_plate_type(vehicle)
+    if vehicle.electric_vehicle?
+      :ev
+    elsif vehicle.antique?
+      :antique
+    else
+      :regular
+    end
+  end
+
+  def determine_fees(vehicle)
+    if determine_plate_type(vehicle) == :regular
+      100
+    elsif determine_plate_type(vehicle) == :ev
+      200
+    else
+      25
+    end
+  end
```

Comment:
I like these helpers! Nice work. Organizationally speaking, I typically put helpers toward the bottom of a class. We don't really get into private methods in Mod 1 (though you are welcome to Google it) but usually helpers are considered private methods. Meaning, we won't use them outside of the class where they exist. Often you'll see private methods at the bottom of a class. It's not absolutely necessary as there are a couple ways to denote methods as private, but it's the most common organization structure I see.